File: MinimumInSortedArray.java
// Time Complexity :o(logn)
// Space Complexity :o(1)
// Did this code successfully run on Leetcode :Yess
// Any problem you faced while coding this :i am not able to optimize it more.
class Solution {
    public int findMin(int[] nums) {
        int n=nums.length;
        int low=0;
        int high=n-1;
        int min=Integer.MAX_VALUE;
        while(low<=high){
            int mid= low+(high-low)/2;
            if(nums[low]<=nums[high]){
                min=Math.min(min, nums[low]);
                return min;
            }
            
            if(nums[low]<=nums[mid]){
                min=Math.min(min, nums[low]);
                low=mid+1;     
            }  
            else{
                min=Math.min(min, nums[mid]);
                high=mid-1;
            }
            
        }
        return min;
    }
}

File: MinimumInSortedArray.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int low = 0; 
        int high = nums.size()-1; 

        if(nums.size() == 1) return nums[0];

        if(nums[nums.size()-1] > nums[0]) return nums[0];

        while(low <= high) {
            int mid = (low+high)/2; 
            if(nums[mid] >= nums[0]) low = mid+1; 
            else high = mid-1; 
        }

        return nums[low]; 
    }
};